trie/utils/verkle.go

```diff
@@ -279,22 +275,27 @@ func GetTreeKeyStorageSlotWithEvaluatedAddress(evaluated *verkle.Point, storageK
 }
 
 func GetTreeKeyStorageSlotTreeIndexes(storageKey []byte) (*uint256.Int, byte) {
-	// Note that `pos` must be a big.Int and not a uint256.Int, because the subsequent
-	// arithmetics operations could overflow. (e.g: imagine if storageKey is 2^256-1)
-	pos := new(big.Int).SetBytes(storageKey)
-	if pos.Cmp(codeStorageDeltaBig) < 0 {
-		pos.Add(headerStorageOffsetBig, pos)
-	} else {
-		pos.Add(mainStorageOffsetBig, pos)
-	}
-	treeIndex, overflow := uint256.FromBig(big.NewInt(0).Div(pos, verkleNodeWidthBig))
-	if overflow { // Must never happen considering the EIP definition.
-		panic("tree index overflow")
+	var pos uint256.Int
+	pos.SetBytes(storageKey)
+
+	// If the storage slot is in the header, we need to add the header offset.
+	if pos.Cmp(codeStorageDelta) < 0 {
+		// This addition is always safe; it can't ever overflow since pos<codeStorageDelta.
+		pos.Add(HeaderStorageOffset, &pos)
+
+		// In this branch, the tree-index is zero since we're in the account header,
+		// and the sub-index is the LSB of the modified storage key.
+		return zero, byte(pos[0] & 0xFF)
+
 	}
-	// calculate the sub_index, i.e. the index in the stem tree.
-	// Because the modulus is 256, it's the last byte of treeIndex
-	posBytes := pos.Bytes()
-	subIndex := posBytes[len(posBytes)-1]
+	// If the storage slot is in the main storage, we need to add the main storage offset.
 
-	return treeIndex, subIndex
+	// We first divide by VerkleNodeWidth to create room to avoid an overflow next.
+	pos.Rsh(&pos, uint(VerkleNodeWidthLog2))
+	// We add mainStorageOffset/VerkleNodeWidth which can't overflow.
+	pos.Add(&pos, mainStorageOffsetLshVerkleNodeWidth)
```

Comment:
I was quite explicit in the comments, but adding a note here.

The TL;DR for this case is that we need to do `(MAIN_STORAGE_OFFSET + storage_key)/VERKLE_NODE_WIDTH`
The detail is that we avoid doing `MAIN_STORAGE_OFFSET + storage_key`, since that would overflow in a `uint256`; so we do it in parts.

First we do `storage_key/VERKLE_NODE_WIDTH`, and then we add `mainStorageOffsetLshVerkleNodeWidth` which is a constant literally `MAIN_STORAGE_OFFSET/VERKLE_NODE_WIDTH`.

Note doing this in part is correct since `MAIN_STORAGE_OFFSET` is a multiple of `VERKLE_NODE_WIDTH`, we we aren't "loosing precision" in the VERKLE_NODE_WIDTH lower bits having "carries".

This is similar to doing something like: `(100 + 12) / 10` which should be `11`. In this case it's correct doing `100/10+12/10` since also yields `11`. 

But note that if we had to do something like `(109+12)/10` it yields 12. But if we try doing `109/10+12/10` that is _incorrect_ since it yields `11`!